src/Microsoft.OData.Core/Metadata/EdmLibraryExtensions.cs

```diff
@@ -252,19 +254,135 @@ internal static IEnumerable<IEdmOperation> FilterBoundOperationsWithSameTypeHier
         }
 
         /// <summary>
-        /// Filters the operations by parameter names.
+        /// Filters the operations by parameter count.
         /// </summary>
         /// <param name="functions">The operations.</param>
-        /// <param name="parameters">The list of non-binding parameter names to match.</param>
-        /// <param name="caseInsensitive">Whether case insensitive.</param>
+        /// <param name="parameterCount">The count of non-binding parameter names to match.</param>
         /// <returns>The best matching operations based on parameters.</returns>
-        internal static IEnumerable<IEdmOperation> FindBestOverloadBasedOnParameters(this IEnumerable<IEdmOperation> functions, IEnumerable<string> parameters, bool caseInsensitive = false)
+        internal static IEnumerable<IEdmOperation> FindBestOverloadBasedOnParameterCount(this IEnumerable<IEdmOperation> functions, int parameterCount)
         {
             // The best match out of a list of candidates is the one that has the same number of (non-binding) parameters as specified.
-            IEnumerable<IEdmOperation> exactMatches = functions.Where(f => f.Parameters.Count() == parameters.Count() + (f.IsBound ? 1 : 0));
+            IEnumerable<IEdmOperation> exactMatches = functions.Where(f => f.Parameters.Count() == parameterCount + (f.IsBound ? 1 : 0));
             return exactMatches.Count() > 0 ? exactMatches : functions;
         }
 
+        /// <summary>
+        /// Filters the operations by parameter count.
+        /// </summary>
+        /// <param name="functions">The operations.</param>
+        /// <param name="parameterCount">The count of non-binding parameter names to match.</param>
+        /// <returns>The best matching operations based on parameters.</returns>
+        internal static IList<IEdmOperation> FindBestOverloadBasedOnParameterCount(this IList<IEdmOperation> functions, int parameterCount)
+        {
+            IList<IEdmOperation> exactMatches = null;
+
+            // The best match out of a list of candidates is the one that has the same number of (non-binding) parameters as specified.
+            for (int i = 0; i < functions.Count; i++)
+            {
+                if (functions[i].Parameters.Count() == parameterCount + (functions[i].IsBound ? 1 : 0))
+                {
+                    if (exactMatches == null)
+                    {
+                        exactMatches = new List<IEdmOperation>();
+                    }
+
+                    exactMatches.Add(functions[i]);
+                }
+            }
+
+            return exactMatches == null ? functions : exactMatches;

```

Comment:
>? functions  [](start = 40, length = 12)

why return all when no exact matches?

Reply:
Keeping the behavior same as the earlier function call. 

---
In reply to: [370381493](https://github.com/OData/odata.net/pull/1602#discussion_r370381493) [](ancestors = 370381493)